services/121-service/test/payment/do-payment-fsp-onafriq.test.ts

```diff
@@ -0,0 +1,201 @@
+      });
+      const paymentReferenceIds = [registrationOnafriq.referenceId];
+
+      // Act
+      const doPaymentResponse = await doPayment({
+        programId,
+        paymentNr: payment,
+        amount,
+        referenceIds: paymentReferenceIds,
+        accessToken,
+      });
+
+      await waitForPaymentTransactionsToComplete({
+        programId,
+        paymentReferenceIds,
+        accessToken,
+        maxWaitTimeMs: 4_000,
+        completeStatusses: Object.values(TransactionStatusEnum),
+      });
+
+      // Assert
+      const getTransactionsBody = await getTransactions({
+        programId,
+        paymentNr: payment,
+        registrationReferenceId: registrationOnafriq.referenceId,
+        accessToken,
+      });
+
+      expect(doPaymentResponse.status).toBe(HttpStatus.ACCEPTED);
+      expect(doPaymentResponse.body.applicableCount).toBe(
+        paymentReferenceIds.length,
+      );
+      expect(getTransactionsBody.body[0].status).toBe(
+        TransactionStatusEnum.error,
+      );
+      expect(getTransactionsBody.body[0].errorMessage).toBe(
+        'Error: 101 - Rejected - Generic mock error on request',
+      );
+    });
+
+    it('should give error via callback based on magic phonenumber', async () => {
+      // Arrange
+      registrationOnafriq.phoneNumber = '24300000001'; // this magic number is configured in mock to return an error on callback
+      await importRegistrations(programId, [registrationOnafriq], accessToken);
+      await awaitChangeRegistrationStatus({
+        programId,
+        referenceIds: [registrationOnafriq.referenceId],
+        status: RegistrationStatusEnum.included,
+        accessToken,
+      });
+      const paymentReferenceIds = [registrationOnafriq.referenceId];
+
+      // Act
+      const doPaymentResponse = await doPayment({
+        programId,
+        paymentNr: payment,
+        amount,
+        referenceIds: paymentReferenceIds,
+        accessToken,
+      });
+
+      // wait for non-waiting transactions only, to make sure callback came in
+      await waitForPaymentTransactionsToComplete({
+        programId,
+        paymentReferenceIds,
+        accessToken,
+        maxWaitTimeMs: 4_000,
+        completeStatusses: [
+          TransactionStatusEnum.success,
+          TransactionStatusEnum.error,
+        ],
+      });
+
+      // Assert
+      const getTransactionsBody = await getTransactions({
+        programId,
+        paymentNr: payment,
+        registrationReferenceId: registrationOnafriq.referenceId,
+        accessToken,
+      });
+
+      expect(doPaymentResponse.status).toBe(HttpStatus.ACCEPTED);
+      expect(doPaymentResponse.body.applicableCount).toBe(
+        paymentReferenceIds.length,
+      );
+      expect(getTransactionsBody.body[0].status).toBe(
+        TransactionStatusEnum.error,
+      );
+      expect(getTransactionsBody.body[0].errorMessage).toBe(
+        'Error: ER103 - Mock error on callback',
+      );
```

Comment:
```suggestion
      expect(getTransactionsBody.body[0].errorMessage).toMatchSnapshot()
```
I like to use toMatchSnapshot for error messages. Because the Ux copy of the error message is often updated in the 121-service code and than you can update the test by running it with -- --u instead of maintaining the ux copy in 2 places.